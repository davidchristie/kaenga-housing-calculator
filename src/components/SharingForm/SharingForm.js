import React, { Component, PropTypes } from 'react'
import { ControlLabel, FormGroup } from 'react-bootstrap'
import { Field, reduxForm } from 'redux-form'

import GuestRoomsField from './GuestRoomsField'
import EntertainmentField from './EntertainmentField'
import LaundriesField from './LaundriesField'

class SharingForm extends Component {
  render () {
    const { handleSubmit } = this.props
    return (
      <form onSubmit={handleSubmit}>
        <FormGroup>
          <ControlLabel>Are you willing to share:</ControlLabel>
          <Field component={LaundriesField} name='laundries' />
          <Field component={GuestRoomsField} name='guestRooms' />
          <Field component={EntertainmentField} name='entertainment' />
        </FormGroup>
      </form>
    )
  }
}

SharingForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default reduxForm({
  form: 'sharing'
})(SharingForm)
