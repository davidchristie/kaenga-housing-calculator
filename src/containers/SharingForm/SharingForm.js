import { reduxForm } from 'redux-form'

import SharingForm from '../../components/SharingForm'

export default reduxForm({
  form: 'sharing'
})(SharingForm)
