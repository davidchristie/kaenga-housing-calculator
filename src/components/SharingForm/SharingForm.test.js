import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import SharingForm from './SharingForm'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<SharingForm />)).toMatchSnapshot()
})
