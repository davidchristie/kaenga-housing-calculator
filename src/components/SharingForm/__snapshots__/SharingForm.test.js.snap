exports[`test matches snapshot 1`] = `
<Connect(Form(SharingForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="sharing"
  getFormState={[Function]}
  initialValues={undefined}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false} />
`;
